source/extensions/clusters/common/logical_host.h

```diff
@@ -61,6 +61,14 @@ class LogicalHost : public HostImpl {
     absl::ReaderMutexLock lock(&address_lock_);
     return HostImpl::address();
   }
+
+  const std::pair<Network::Address::InstanceConstSharedPtr,
+                  const std::vector<Network::Address::InstanceConstSharedPtr>>
```

Comment:
readability nit: make a using declaration for this vector, and another one for the pair, and you can use them in the impl as well.